### PR TITLE
test images: Adds Windows Server 2022 to the BASEIMAGEs (part 2)

### PR DESCRIPTION
**Pull request by claudiubelu**
#### What type of PR is this?

/sig windows
/sig testing

/kind feature
/priority important-soon

#### What this PR does / why we need it:

The Container Images for Windows Server 2022 have been published, and we can start building test images using them, so we can start adding jobs for them.

The image versions for the e2e test images have been bumped in a previous commit [0], but haven't been promoted yet. We don't need to bump them here.

[0] https://github.com/kubernetes/kubernetes/pull/102599
[1] https://github.com/kubernetes/k8s.io/blob/c8b744d34f4f017ff6989557a2acc37ed2d3f1be/k8s.gcr.io/images/k8s-staging-e2e-test-images/images.yaml#L1

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by jsturtevant** [comment]
xref #104451

File: test/images/agnhost/README.md [hunk 1]
```diff
@@ -643,6 +643,7 @@ The Windows `agnhost` image includes a `nc` binary that is 100% compliant with i
 
 ## Image
 
-The image can be found at `k8s.gcr.io/e2e-test-images/agnhost:2.32` for both Linux and
+The image can be found at `k8s.gcr.io/e2e-test-images/agnhost:2.33` for both Linux and
```
Comment (mkumatag) [inline test/images/agnhost/README.md:646]: not part of this PR but we need to find some other way to avoid mentioning the images with versions here, because this will get outdated anytime or find some way to update this version when a file changes.

**Comment by claudiubelu** [comment]
@mkumatag can we get these in?

**Review by mkumatag — APPROVED** [review]
/lgtm

**Comment by jsturtevant** [comment]
/lgtm 
for sig-windows

Thanks @claudiubelu!

**Comment by aojea** [comment]
> /lgtm
> for sig-windows
> 
> Thanks @claudiubelu!

/hold cancel
